compiler/parser/RdrHsSyn.hs

```diff
@@ -2673,6 +2678,7 @@ mkTemplateClassInstanceMethods conName ValidTemplate{..} =
     templateName = rdrNameToString conName
     this = asPatRecWild "this" conName
     cid = mkVarPat $ mkVarOcc "cid"
+    proxy = mkVarPat $ mkVarOcc "proxy"
```

Comment:
Why do you need the proxy in the arguments here? I would have thought that it should be fine to not match on anything.

Reply:
The type of the builtin is just `Text`, but the method is `toTextTemplateId : Template t => proxy t -> Text`. I've added the variable pattern to match those types, otherwise this raises a type error down the line.

Reply:
I’m not sure I’m following, why can’t you generate a lambda in the LF conversion that binds `proxy` similar to what I did for `toAnyTemplate` https://github.com/digital-asset/daml/pull/3152/files#diff-b0d40c4af4444bad82fcc3d9b35e4ed4R411? Not necessarily saying that is better, I’m mostly curious why this doesn’t work here.

Reply:
as I said, I was mostly curious so if your approach works, I’m fine with keeping the pattern.